docs/source/contributing/contributing-code.rst

```diff
@@ -0,0 +1,90 @@
+Contributing code
+==================================================
+
+Local setup
+----------------------------------
+
+To contribute code, basic knowledge of git is required, if you are unfamiliar
+with the git workflow, checkout `GitHub's tutorial <https://docs.github.com/en/get-started/quickstart/hello-world>`_.
+The following steps allow you to contribute code to DoWhy.
+
+1. Fork the `dowhy main repository <https://github.com/py-why/dowhy>`_
+2. Clone this repository to your local machine using
+
+.. code:: shell
+
+   git clone https://github.com/<YOUR_GITHUB_USERNAME>/dowhy
+
+3. [optional] Create a virtual environment using python venv or conda #check if poetry does this automatically
+
+.. code:: shell
+
+   virtualenv dowhy-env --python=python3.8
+   source dowhy-env/bin/activate
+
+Note this step is not necessary as Poetry will create a new environment by itself if needed.
+
+4. Install DoWhy and it's requirements in your environment.
+By default, Poetry will install DoWhy in interactive mode.
+This way, you can immediately test your changes to the codebase.
+
+.. code:: shell
+
+   cd dowhy
+   pip install --upgrade pip
+   poetry install -E "plotting causalml docs"
```

Comment:
I'd recommend removing the docs extra - those dependencies will likely be moved to dev-dependencies since they're not strictly necessary at runtime.